crates/next-core/src/next_client_reference/ecmascript_client_reference/ecmascript_client_reference_transition.rs

```diff
@@ -71,7 +71,11 @@ impl Transition for NextEcmascriptClientReferenceTransition {
                     .replace("next/dist/esm/", "next/dist/")
                     .into(),
             );
-            Vc::upcast(FileSource::new_with_query(path, *ident_ref.query))
+            Vc::upcast(FileSource::new_with_query_and_fragment(
+                path,
+                (*ident_ref.query.await?).clone(),
```

Comment:
You could write ident_ref.query.owned().await?

Reply:
oh yeah, that is nicer.  ill fix this in the followup